### DISCUSSION TITLE
Show fullpath instead of filename in RichHandler

### DISCUSSION THREAD

**Discussion by lucmos** [Textualize/rich#1818] [Ideas]
In my experience, it would be more beneficial to display the full path instead of the filename in the logs.
For example, it isn't easy to understand which library the log belongs to from only the filename.

From my understanding, it should be enough to change this line:
https://github.com/Textualize/rich/blob/2ba277ac54a153d3030b496698133ec2e4f67e1d/rich/logging.py#L195

What do you think?

I am currently using the `record.name` instead

**Comment by willmcgugan** [comment]
That would work. If you want that, I'd suggest creating a subclass. Generally the path is so long that there wouldn't be any room for the log message.

BTW if you have a terminal that supports hyperlinks you can hover over the path to see the full path.

**Reply by slafs** [reply]
How about adding a new option for showing the name of the logger (from `record.name`) _instead_ of the filename? It's a common practice to name your loggers with `getLogger(__name__)` and name would be shorter than a full path. Name and path options would probably be mutually exclusive, but could have the same link underneath. What do you think? Is this something you'd find a good addition to rich? I can open an issue and a PR later.
